Develop

Dyuzhovsergey opened the pull request.

Dyuzhovsergey committed:
Add in func TestCafeWhenBad struct requestTest

Dyuzhovsergey committed:
Add test func TestCafeCount

Dyuzhovsergey committed:
Add test func TestCafeSearch

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	//"golang.org/x/text/search"
```
лишние комментарии стоит удалять

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
+		req := httptest.NewRequest("GET", url, nil)
+		handler.ServeHTTP(response, req)
+
+		assert.Equal(t, http.StatusOK, response.Code)
```
require

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
+		{count: 0, want: 0},
+		{count: 1, want: 1},
+		{count: 2, want: 2},
+		{count: 100, want: len(cafeList["moscow"])},
```
want: min(100, len(cafeList["moscow"]))

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
+		if tc.want == 0 && len(result[0]) == 0 {
+			result = []string{}
+		}
+		assert.Equal(t, tc.want, len(result), fmt.Sprintf("unexpected cafe count. Count=%d", tc.count))
```
assert.Len

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
 		handler.ServeHTTP(response, req)
 
 		assert.Equal(t, http.StatusOK, response.Code)
+
```
require.Equal

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
+			result = []string{}
+		}
+
+		assert.Equal(t, tc.wantCount, len(result), fmt.Sprintf("unexpected count cafe for query='%s'", tc.search))
```
assert.Len

dirtymew commented on main_test.go:
```diff
@@ -1,50 +1,123 @@
+
+		for _, cafe := range result {
+			ok := strings.Contains(strings.ToLower(cafe), strings.ToLower(tc.search))
+			assert.True(t, ok, fmt.Sprintf("cafe name %s does not contain search query %s", cafe, tc.search))
```
assert.Contains

dirtymew reviewed (approved):
Привет Сергей

хорошая работа - но есть что улучшить см PR

Dyuzhovsergey committed:
Add assert.Len() instesd assert.Equal()

Dyuzhovsergey commented:
Приветствую, Антон!
Я все поправил!
Спасибо за обратную связь!
Про asesert.Contains совсем забыл

пт, 4 июл. 2025 г., 16:03 Anton Sattarov ***@***.***>:

> ***@***.**** approved this pull request.
>
> Привет Сергей
>
> хорошая работа - но есть что улучшить см PR
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185315581>
> :
>
> >  	"net/http"
>  	"net/http/httptest"
>  	"strings"
>  	"testing"
>
>  	"github.com/stretchr/testify/assert"
> +	//"golang.org/x/text/search"
>
> лишние комментарии стоит удалять
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185316554>
> :
>
> >  		response := httptest.NewRecorder()
> -		req := httptest.NewRequest("GET", v, nil)
> +		req := httptest.NewRequest("GET", url, nil)
> +		handler.ServeHTTP(response, req)
> +
> +		assert.Equal(t, http.StatusOK, response.Code)
>
> require
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185317125>
> :
>
> >  	handler := http.HandlerFunc(mainHandle)
>
> -	requests := []string{
> -		"/cafe?count=2&city=moscow",
> -		"/cafe?city=tula",
> -		"/cafe?city=moscow&search=ложка",
> +	requests := []struct {
> +		count int
> +		want  int
> +	}{
> +		{count: 0, want: 0},
> +		{count: 1, want: 1},
> +		{count: 2, want: 2},
> +		{count: 100, want: len(cafeList["moscow"])},
>
> want: min(100, len(cafeList["moscow"]))
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185317545>
> :
>
> >
> +		result := strings.Split(response.Body.String(), ",")
> +		if tc.want == 0 && len(result[0]) == 0 {
> +			result = []string{}
> +		}
> +		assert.Equal(t, tc.want, len(result), fmt.Sprintf("unexpected cafe count. Count=%d", tc.count))
>
> assert.Len
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185319146>
> :
>
> >  		handler.ServeHTTP(response, req)
>
>  		assert.Equal(t, http.StatusOK, response.Code)
> +
>
> require.Equal
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185319425>
> :
>
> >  		handler.ServeHTTP(response, req)
>
>  		assert.Equal(t, http.StatusOK, response.Code)
> +
> +		result := strings.Split(response.Body.String(), ",")
> +		if tc.wantCount == 0 && len(result[0]) == 0 {
> +			result = []string{}
> +		}
> +
> +		assert.Equal(t, tc.wantCount, len(result), fmt.Sprintf("unexpected count cafe for query='%s'", tc.search))
>
> assert.Len
> ------------------------------
>
> In main_test.go
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#discussion_r2185319768>
> :
>
> >  		handler.ServeHTTP(response, req)
>
>  		assert.Equal(t, http.StatusOK, response.Code)
> +
> +		result := strings.Split(response.Body.String(), ",")
> +		if tc.wantCount == 0 && len(result[0]) == 0 {
> +			result = []string{}
> +		}
> +
> +		assert.Equal(t, tc.wantCount, len(result), fmt.Sprintf("unexpected count cafe for query='%s'", tc.search))
> +
> +		for _, cafe := range result {
> +			ok := strings.Contains(strings.ToLower(cafe), strings.ToLower(tc.search))
> +			assert.True(t, ok, fmt.Sprintf("cafe name %s does not contain search query %s", cafe, tc.search))
>
> assert.Contains
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Dyuzhovsergey/searchCafe/pull/1#pullrequestreview-2987150631>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGBW55PQW3XVE3OHCYCIPST3GZ3QNAVCNFSM6AAAAACAXL3OFKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSOBXGE2TANRTGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
